scripts/valhalla_build_extract.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+import argparse
+from io import BytesIO
+import json
+import logging
+from pathlib import Path
+import struct
+import sys
+import tarfile
+from tarfile import BLOCKSIZE
+from typing import Dict
+
+description = """Builds a tar extract from the tiles in mjolnir.tile_dir to the path specified in mjolnir.tile_extract."""
+
+# "=" prefix means native byte order with standard size and no alignment:
+# https://docs.python.org/3/library/struct.html#byte-order-size-and-alignment
+STRUCT_FORMAT = '=LQ'
+STRUCT_SIZE = struct.calcsize(STRUCT_FORMAT)
+INDEX_FILE = "index.bin"
+TAR_ARCNAME_DIR = Path('valhalla_tiles')
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-c", "--config", help="Absolute or relative path to the Valhalla config JSON.", type=Path)
+parser.add_argument("-v", "--verbosity", help="Accumulative verbosity flags; -v: INFO, -vv: DEBUG", action='count', default=0)
+
+# set up the logger basics
+LOGGER = logging.getLogger(__name__)
+handler = logging.StreamHandler()
+handler.setFormatter(logging.Formatter("%(asctime)s %(levelname)5s: %(message)s"))
+LOGGER.addHandler(handler)
+
+
+def get_padded_tar_size(data_size: int) -> int:
+    """Returns the tar-internal file size, incl. header and NULL padding"""
+    #      <header>    <-data->    <-------------------NULL padding---------------->
+    return BLOCKSIZE + data_size + (BLOCKSIZE - (data_size % BLOCKSIZE or BLOCKSIZE))
+
+
+def get_tile_info(in_path: Path, tiles: dict):
+    """Recursively fills the passed map with tile path & size info"""
+    for p in in_path.iterdir():
+        if p.is_dir():
+            get_tile_info(p, tiles)
+        if p.is_file():
+            tiles[p.resolve()] = get_padded_tar_size(p.stat().st_size)
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+    with open(args.config) as f:
+        config = json.load(f)
+    index_fp: Path = args.config.parent.joinpath(INDEX_FILE)
+    extract_fp: Path = Path(config["mjolnir"]["tile_extract"])
+    tiles_fp: Path = Path(config["mjolnir"]["tile_dir"])
```

Comment:
I'll break most of the logic out of the `__main__` so it's easier to test it. basically only keeping the args parsing